core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -56,7 +60,9 @@ class LogManager(val logDirs: Array[File],
                  scheduler: Scheduler,
                  val brokerState: BrokerState,
                  brokerTopicStats: BrokerTopicStats,
-                 time: Time) extends Logging {
+                 val zkUtils: ZkUtils,
```

Comment:
Another reason to put the logic of zookeeper notification in LogManger is that, currently LogManager is managing which logs are online or offline and thus is the source of truth of offline log directories. Ideally we would like to put the logic of notification in the same place so that the notification is sent if and only if the offline log directories change.